test/Microsoft.ML.Tests/TextLoaderTests.cs

```diff
@@ -795,5 +796,77 @@ public void TestTextLoaderKeyTypeBackCompat()
                 Assert.True(result.Schema[featureIdx].Type is KeyDataViewType keyType && keyType.Count == typeof(uint).ToMaxInt());
             }
         }
+
+        public class IrisNoFields
```

Comment:
Is it possible to do `private class`?

Reply:
If your question is whether the class defining the data can be private, the answer is yes.
If you are asking to make these classes private in the test, I would prefer to keep them public since right above they were defined this way. I would rather be consistent than changing here only.

Reply:
I feel public surface should be as small as possible. It helps both devs (they know which thing not to look for example) and users (user only accepts necessary information) a lot.

Reply:
Ok I understand your point! I will update this to be private.